src/asm/opt.c

```diff
@@ -86,6 +96,43 @@ void opt_Parse(char *s)
 		}
 		break;
 
+	case 'r': {
+		++s; // Skip 'r'
+		while (isblank(*s))
```

Comment:
I see this extra parsing logic exists to support `%` and `$` literals. Currently passing numeric arguments via the command line does not support those, so I'm not sure `OPT` needs to either. If it does, this could be extracted into a custom `read_asm_number` function, even if so far it's only used here.

```c
		++s; // Skip 'r'

		long newDepth = read_asm_num(s);

		if (newDepth < 0) {
			error("Invalid argument to option 'r' (\"%s\")\n", s);
		} else {
			opt_R((unsigned long)newDepth);
		}
		break;
```

```c
long read_asm_number(char const *s) {
	while (isblank(*s))
		++s; // Skip leading whitespace

	char const *digits = s;
	int base = 10;

	if (*digits == '\0') {
		return -1;
	} else if (*digits == '$') {
		base = 16;
		++digits;
	} else if (*digits == '&') {
		base = 8;
		++digits;
	} else if (*digits == '%') {
		base = 2;
		++digits;
	}

	errno = 0;
	char *endptr;
	unsigned long value = strtoul(digits, &endptr, base);

	if (digits[0] == isspace(digits[0]) || digits[0] == '\0' || *endptr != '\0' || errno == ERANGE) {
		return -1;
	}

	return value;
}
```

Reply:
Actually, neither the current code not the example `read_asm_number` code would yet handle custom binary digits. I think we should just support the same values as the command line, and use `strtoul` without pre-parsing. People can always do `OPT r {d:depth}` if they need a non-literal arg value in base 10.